### PR TITLE
ICSAAS-400 update search tab cmp

### PR DESCRIPTION
**Pull request by igorsimovski**
- add search template field in dialog
 - update FulltextSearchService interface
  - add template parameter in base method
 - update SearchTabModel sling model
  - bind config value from jcr
  - pass value in FulltextSearchService::getResults
  - update unit test

### CODE REVIEW

File: core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/DefaultFulltextSearchService.java [hunk 1]
```diff
@@ -87,19 +90,31 @@ private String createQueryString(
...
-                                        .query(FacetsQuery.builder()
-                                                          .fields(CollectionUtils.emptyIfNull(facets))
-                                                          .build())
-                                        .build()
-                                        .getQueryString();
+            Set<String> facets,
+            String template) {
+        GetQueryStringConstructor.GetQueryStringConstructorBuilder builder =
+                GetQueryStringConstructor.builder()
+                                         .query(new TermQuery(
```
Comment (nhirrle) [inline core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/DefaultFulltextSearchService.java:97]: could you optimize the linebreaks heew? before it was more readable

**Comment by nhirrle** [comment]
Have you done a End to End test? Ideally an Integration Test could be peformed to see if the result was updated

**Review by nhirrle — COMMENTED** [review]
left few comments with suggested changes

**Comment by igorsimovski** [comment]
> Have you done a End to End test? Ideally an Integration Test could be peformed to see if the result was updated

test added

**Pull request merged**